pkg/controller/utils/component.go

```diff
@@ -66,14 +66,19 @@ type componentHandler struct {
 	log    logr.Logger
 }
 
-func (c componentHandler) createOrUpdateObject(ctx context.Context, obj client.Object, metaObj metav1.Object, osType rmeta.OSType) error {
+func (c componentHandler) createOrUpdateObject(ctx context.Context, obj client.Object, osType rmeta.OSType) error {
+	om, ok := obj.(metav1.ObjectMetaAccessor)
+	if !ok {
+		return fmt.Errorf("Object is not ObjectMetaAccessor")
```

Comment:
Do we perhaps need to log/learn more about this object? If this can happen it looks indicative of a bug. Won't this error keep repeating on any reconcile?

Reply:
Yeah it would repeat. I copied this from before I just moved it. I don't imagine this is actually needed. It might be something to make the linter happy, to check that casting was successful.